src/stirling/source_connectors/socket_tracer/uprobe_symaddrs.cc

```diff
@@ -687,18 +704,32 @@ StatusOr<struct openssl_symaddrs_t> OpenSSLSymAddrs(RawFptrManager* fptrManager,
   PL_ASSIGN_OR_RETURN(uint32_t openssl_fix_sub_version,
                       OpenSSLFixSubversionNum(fptrManager, openssl_lib, pid));
 
-  switch (openssl_fix_sub_version) {
-    case 0:
-      symaddrs.RBIO_num_offset = kOpenSSL_1_1_0_RBIO_num_offset;
-      break;
-    case 1:
-      symaddrs.RBIO_num_offset = kOpenSSL_1_1_1_RBIO_num_offset;
-      break;
-    default:
-      // Supported versions are checked in function OpenSSLFixSubversionNum(),
-      // should not fall through to here, ever.
-      DCHECK(false);
-      return error::Internal("Unsupported openssl_fix_sub_version: $0", openssl_fix_sub_version);
+  if (!IsBoringSSL(openssl_lib)) {
+    switch (openssl_fix_sub_version) {
+      case 0:
+        symaddrs.RBIO_num_offset = kOpenSSL_1_1_0_RBIO_num_offset;
+        break;
+      case 1:
+        symaddrs.RBIO_num_offset = kOpenSSL_1_1_1_RBIO_num_offset;
+        break;
+      default:
+        // Supported versions are checked in function OpenSSLFixSubversionNum(),
+        // should not fall through to here, ever.
+        DCHECK(false);
+        return error::Internal("Unsupported openssl_fix_sub_version: $0", openssl_fix_sub_version);
+    }
+  } else {
+    switch (openssl_fix_sub_version) {
+      case 1:
+        symaddrs.RBIO_num_offset = kBoringSSL_1_1_1_RBIO_num_offset;
+        symaddrs.SSL_rbio_offset = kBoringSSL_RBIO_offset;
```

Comment:
So boringssl has no 1.0 version? Could you link to boringssl offical doc (if such statement exists)?

Reply:
The [PORTING.md](https://github.com/google/boringssl/blob/0cc51a793eef6b5295b9e0de8aafb1d87a39e210/PORTING.md) docs in boringssl explains that it was initially derived from 1.0.2 and that `OPENSSL_VERSION_NUMBER` will match the version it targets. That define is what [OpenSSL_version_num returns](https://github.com/google/boringssl/blob/a94c267787df4ac0517456231ebc7be40442895f/crypto/crypto.c#L214-L216).

> BoringSSL's OPENSSL_VERSION_NUMBER matches the OpenSSL version it targets. Version checks for OpenSSL should ideally work as-is in BoringSSL. BoringSSL also defines upstream's OPENSSL_NO_* feature macros corresponding to removed features. If the preprocessor is needed, use these version checks or feature macros where possible, especially when patching third-party projects. Such patches are more generally useful to OpenSSL consumers and thus more appropriate to send upstream.
> 
> [ ... ]
> 
> BoringSSL was originally derived from OpenSSL 1.0.2 but now targets OpenSSL 1.1.0. Some newer APIs may be missing but can be added on request. (Not all projects have been ported to OpenSSL 1.1.0, so BoringSSL also remains largely compatible with OpenSSL 1.0.2.)

I've updated the comment to include a link to these docs and added some more detail.

Reply:
Please leave a TODO() to investigate the boringssl 1.0 version's offsets. You can assign that to me.

As you mentioned, the scope of netty probably does not need to address the 1.0 version for now.

Reply:
Sounds good. I've created https://github.com/pixie-io/pixie/issues/588 and linked it in the TODO I created for you.